### PR TITLE
Adding new versions of bcpkix, hbase-shaded-client and velocity

### PR DESCRIPTION
**Pull request by ruwany**
hbase-shaded-client imports javax.servlet.* package of range [2.2.1,3.0.0), when used like this, the classification encompasses javax.servlet.jsp.* packages as well, which are of a different version range E.g. [2.2.0,2.2.3). This causes issues at feature installation. A new version has been added with the fix to resolve this.

bcpkix and bcmail 1.49 versions are not compatible with bcprov 1.52, it requires version 1.52. This PR adds the new 1.52 versions.

Velocity jar is not OSGI compatible, and without dynamic-dependencies it cannot load classes on the fly, this PR adds an OSGIfied velocity 1.7 with dynamic dependencies.

### CODE REVIEW

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<!--
+ ~ Copyright (c) 2005-2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: year should be 2016

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    <groupId>org.wso2.orbit.org.bouncycastle</groupId>
+    <artifactId>bcmail-jdk15on</artifactId>
+    <packaging>bundle</packaging>
+    <name>bcmail</name>
+    <version>1.52.0.wso2v1</version>
+    <description>
+        This bundle will represent org.bouncycastle
+    </description>
+    <url>http://wso2.org</url>
+    <repositories>
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: why we need repository section

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                <updatePolicy>daily</updatePolicy>
+                <checksumPolicy>fail</checksumPolicy>
+            </releases>
+        </repository>
+    </repositories>
+    <dependencies>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcmail-jdk15on</artifactId>
+            <version>1.52</version>
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: all the dependencies should be optional

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,98 @@
+<!--
+ ~ Copyright (c) 2005-2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ ~
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: distribution section is missing

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.bouncycastle.mail.smime.*,
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: Better to include export version

File: bcmail/1.52.0.wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                            org.bouncycastle.pqc.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.ocsp.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.math.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.jce.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.jcajce.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.i18n.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.crypto.*;version="${bouncycastle.import.version.range}",
+                            org.bouncycastle.asn1.*;version="${bouncycastle.import.version.range}"
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (thusithathilina) [inline bcmail/1.52.0.wso2v1/pom.xml]: Do we need dynamic imports?

File: bcpkix/1.52.0.wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,149 @@
+<!--
+ ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thusithathilina) [inline bcpkix/1.52.0.wso2v1/pom.xml]: year should be 2016

File: bcpkix/1.52.0.wso2v1/pom.xml [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    <url>http://wso2.org</url>
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+
+        <snapshotRepository>
```
Comment (thusithathilina) [inline bcpkix/1.52.0.wso2v1/pom.xml]: remove snapshot repo

File: bcpkix/1.52.0.wso2v1/pom.xml [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            <id>wso2.snapshots</id>
+            <name>Apache Snapshot Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/snapshots/</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+    <repositories>
```
Comment (thusithathilina) [inline bcpkix/1.52.0.wso2v1/pom.xml]: Do we need this?

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml [hunk 10]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thusithathilina) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: year should be 2016

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    </repositories>
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+
+        <snapshotRepository>
```
Comment (thusithathilina) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: remove snapshot

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+    <artifactId>hbase-shaded-client</artifactId>
+    <version>1.2.1.wso2v2</version>
+    <packaging>bundle</packaging>
+    <name>WSO2 Carbon Orbit - Apache HBase shaded client</name>
+    <description>
+        This bundle will export packages from the Apache HBase shaded client
+    </description>
+    <url>http://wso2.org</url>
+
+    <repositories>
```
Comment (thusithathilina) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: Do we need this?

File: velocity/1.7.0.wso2v1/pom.xml [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+                        <Private-Package>
+                        </Private-Package>
+                        <Import-Package>
+                            com.werken.xpath;resolution:=optional,
+                            javax.naming,
+                            javax.servlet;resolution:=optional,
+                            javax.servlet.http;resolution:=optional,
+                            javax.sql,
+                            org.apache.commons.collections,
+                            org.apache.commons.collections.map,
```
Comment (thusithathilina) [inline velocity/1.7.0.wso2v1/pom.xml]: Don't we need specific version ranges?

File: velocity/1.7.0.wso2v1/pom.xml [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+    </distributionManagement>
+
+    <repositories>
```
Comment (thusithathilina) [inline velocity/1.7.0.wso2v1/pom.xml]: Do we need this?

File: bcmail/1.52.0.wso2v1/pom.xml (hunk 1 above)
Comment (ruwany) [inline bcmail/1.52.0.wso2v1/pom.xml]: Fixed

File: bcmail/1.52.0.wso2v1/pom.xml (hunk 4 above)
Comment (ruwany) [inline bcmail/1.52.0.wso2v1/pom.xml]: Fixed

File: bcmail/1.52.0.wso2v1/pom.xml (hunk 2 above)
Comment (ruwany) [inline bcmail/1.52.0.wso2v1/pom.xml]: Removed

File: bcmail/1.52.0.wso2v1/pom.xml (hunk 5 above)
Comment (ruwany) [inline bcmail/1.52.0.wso2v1/pom.xml]: Added Export version

File: bcpkix/1.52.0.wso2v1/pom.xml (hunk 7 above)
Comment (ruwany) [inline bcpkix/1.52.0.wso2v1/pom.xml]: Fixed

File: bcpkix/1.52.0.wso2v1/pom.xml (hunk 9 above)
Comment (ruwany) [inline bcpkix/1.52.0.wso2v1/pom.xml]: Removed

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml (hunk 12 above)
Comment (ruwany) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: Fixed

File: velocity/1.7.0.wso2v1/pom.xml (hunk 14 above)
Comment (ruwany) [inline velocity/1.7.0.wso2v1/pom.xml]: Removed

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml (hunk 10 above)
Comment (ruwany) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: Fixed

File: bcpkix/1.52.0.wso2v1/pom.xml (hunk 8 above)
Comment (ruwany) [inline bcpkix/1.52.0.wso2v1/pom.xml]: Removed

File: velocity/1.7.0.wso2v1/pom.xml (hunk 13 above)
Comment (ruwany) [inline velocity/1.7.0.wso2v1/pom.xml]: Fixed

File: bcmail/1.52.0.wso2v1/pom.xml (hunk 6 above)
Comment (ruwany) [inline bcmail/1.52.0.wso2v1/pom.xml]: bcmail is no longer needed. Hence removed this from commit.

**Comment by ruwany** [comment]
Added jscep 2.3.0 to orbit as it needed to comply with bouncycastle 1.52 dependencies.

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.wso2.orbit.org.apache.hadoop.hbase</groupId>
```
Comment (thusithathilina) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: Dependency groupid is org.apache.hbase therefore this should be org.wso2.orbit.org.apache.hbase

File: jscep/2.3.0.wso2v1/pom.xml [hunk 16]
```diff
@@ -0,0 +1,74 @@
+<!--
+ ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thusithathilina) [inline jscep/2.3.0.wso2v1/pom.xml]: year should be 2016

File: jscep/2.3.0.wso2v1/pom.xml [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.jscep.*;version="2.3.0",
+                        </Export-Package>
+                        <Import-Package>
+                            org.apache.commons.io.*;version="[2.4.0, 2.5.0)",
+                            !org.jscep.*,
```
Comment (thusithathilina) [inline jscep/2.3.0.wso2v1/pom.xml]: Better to include !pkgs in the beginning

File: hbase-shaded-client/1.2.1.wso2v2/pom.xml (hunk 15 above)
Comment (ruwany) [inline hbase-shaded-client/1.2.1.wso2v2/pom.xml]: Fixed.

File: jscep/2.3.0.wso2v1/pom.xml (hunk 16 above)
Comment (ruwany) [inline jscep/2.3.0.wso2v1/pom.xml]: Fixed

File: jscep/2.3.0.wso2v1/pom.xml (hunk 17 above)
Comment (ruwany) [inline jscep/2.3.0.wso2v1/pom.xml]: Fixed. and updated plugin version as well.

**Pull request merged**